bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/CaddxProperty.java

```diff
@@ -0,0 +1,128 @@
+ */
+@NonNullByDefault
+public class CaddxProperty {
+    // private
+    private final String name;
+    private final CaddxPropertyType type; // 'Int', 'String', 'Bit'
+    private final int byteFrom;
+    private final int byteLength;
+    private final int bitFrom;
+    private final int bitLength;
+    private final boolean external;
+    private final String id;
+
+    // Constructor
+    public CaddxProperty(String id, int byteFrom, int byteLength, int bitFrom, int bitLength, CaddxPropertyType type,
+            String name, boolean external) {
+        this.id = id;
+        this.name = name;
+        this.type = type;
+        this.byteFrom = byteFrom;
+        this.byteLength = byteLength;
+        this.bitFrom = bitFrom;
+        this.bitLength = bitLength;
+        this.external = external;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public CaddxPropertyType getType() {
+        return type;
+    }
+
+    public boolean getExternal() {
+        return external;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getValue(byte[] message) {
+        int mask;
+        int val;
+
+        switch (type) {
+            case INT:
+                if (bitFrom == 0 && bitLength == 0) {
+                    mask = 255;
+                    val = message[byteFrom - 1] & mask;
+                } else {
+                    mask = ((1 << ((bitLength - bitFrom))) - 1) << bitFrom;
+                    val = (message[byteFrom - 1] & mask) >> bitFrom;
+                }
+
+                return Integer.toString(val);
+            case STRING:
+                byte[] str = Arrays.copyOfRange(message, byteFrom - 1, byteFrom + byteLength);
+                return new String(str);
+            case BIT:
+                return (((message[byteFrom - 1] & (1 << bitFrom)) > 0) ? "true" : "false");
+            default:
+                throw new IllegalArgumentException("type is unknown.");
+        }
+    }
+
+    public String toString(byte[] message) {
+        int mask;
+        int val;
+
+        switch (type) {
+            case INT:
+                if (bitFrom == 0 && bitLength == 0) {
+                    mask = 255;
+                    val = message[byteFrom - 1];
+                } else {
+                    mask = ((1 << ((bitLength - bitFrom) + 1)) - 1) << bitFrom;
+                    val = (message[byteFrom - 1] & mask) >> bitFrom;
+                }
+
+                return name + ": " + String.format("%2s", Integer.toHexString(val)) + " - " + Integer.toString(val)
+                        + " - " + ((val >= 32 && val <= 'z') ? ((char) val) : "-");
+            case STRING:
+                StringBuilder sb = new StringBuilder();
+
+                byte[] a = Arrays.copyOfRange(message, byteFrom - 1, byteFrom + byteLength);
+                sb.append(name);
+                sb.append(": ");
+                sb.append(new String(a));
```

Comment:
It would be better to explicitly state the Charset you want to use for the String since the default one varies from system to system.

Reply:
Here i face a small issue.
The panel uses the US ASCII charset, but to display 10 specific Greek characters it uses a custom encoding. (The rest of the Greek characters are the same as in US ASCII)
I was thinking of writing a CharsetProvider to do the decode, but i thought this would be overkill and i ended up to do the mapping of the string just before updating the zone name channel.
ThingHandlerZone.java lines 56:95.
I moved these lines to a utility method in CaddxProperty.java